### ISSUE TITLE
API for adding boat types

### ISSUE THREAD

**Issue by SollyW** [FabricMC/fabric#688]
Boats are a clusterfuck that uses an enum (and ordinals) everywhere to identify the boat type. Even extending the vanilla BoatEntity class would be difficult, as the enum and/or ordinals are also used everywhere that needs the boat type. 

[Terraform adds boats](https://github.com/TerraformersMC/Terraform/blob/1.16/src/main/java/com/terraformersmc/terraform/entity/TerraformBoatEntity.java), but it's extremely hacky.
There really needs to be a way to do this - at this point I won't be adding boats to my mod simply due to there being no viable way for me to do so - but should it go in fapi? I'm not sure.

**Comment by Earthcomputer** [comment]
I think a real fix would be to add a way to add to enums with fabric's mixin fork, or something similar.

**Comment by Solly-W** [comment]
Yes, it would be. If you're taking that route mm would work as well, though.

**Comment by bluebear94** [comment]
That would break the case when e.g. the client has a mod that adds boats but is connected to a server without that mod, or when the client and the server has mods loaded in different orders (if a certain order isn't guaranteed when adding to enums).

**Comment by haykam821** [comment]
One third-party solution is to use [Terraform](https://github.com/TerraformersMC/Terraform)'s API to add boat types.

**Comment by cassiancc** [comment]
Terraform's API is sufficient enough and actively used on versions prior to 1.21.2, where Mojang split boats into their multiple entities, which makes adding new boats trivial.
